cassandra/pool.py

```diff
@@ -385,19 +399,7 @@ def borrow_connection(self, timeout, routing_key=None):
             else:
                 shard_id = random.randint(0, self.host.sharding_info.shards_count - 1)
 
-        if shard_id not in self._connections:
-            for _ in range(self._connections[0].sharding_info.shards_count * 2):
-                c = self._session.cluster.connection_factory(self.host.endpoint)
-                if c.shard_id not in self._connections.keys():
-                    log.debug("new connection create for shard_id=%i", c.shard_id)
-                    self._connections[c.shard_id] = c
-                    if self._keyspace:
-                        self._connections[c.shard_id].set_keyspace_blocking(self._keyspace)
-
-                if c.shard_id == shard_id:
-                    break
-
-        conn = self._connections[shard_id]
+        conn = self._connections.get(shard_id)
```

Comment:
I think here you rely on the fact that you HAVE a connection for each shard.
Therefore you need to either ensure this above - otherwise people will start getting those `NoConnectionsAvailable()` exceptions and get angry at you. ;)

Reply:
I'm married, I'm used for people get angry at me... 

yeah, make sure in the constructor of the pool might be better idea